addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -71,22 +81,26 @@ class DataEgressService extends Service {
     const folderName = `${environment.id}/`;
 
     try {
-      s3Service.createFolder(bucketName, folderName);
+      s3Service.createPath(bucketName, folderName);
     } catch (error) {
       throw this.boom.badRequest(`Error in creating egress store:${folderName} in bucket: ${bucketName}`, true);
     }
 
     // prepare info for ddb and update egress store info
     const egressStoreId = uuid();
+    const creationTime = new Date().toISOString;
     const dbObject = {
       id: egressStoreId,
```

Comment:
This just caught my attention. Seems like this API is not idempotent (i.e. for multiple calls for the same environment we will keep creating new entries in DDB), and this can happen in case of retries due to bad connection or any other reason. We should fix this. Can we just use something like `egress-{environmentId}` as the identifier to make it easer.

Note that while retrieving the entry, if we are changing the right one. Best to just ensure we have one entry instead of doing a scan an changing one entry.